### ISSUE TITLE
[Accessibility]: Focus does not change and there is no spoken feedback when navigating to a subpage of settings

### ISSUE THREAD

**Issue by courier-new** [sourcegraph/sourcegraph#34481]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34073

### Problem description

From user settings, when I use my screen reader to click an item from the left settings menu (e.g. "Profile", the item remains focused and there is no feedback that the settings content on the right side has changed.

This issue was uncovered while auditing a specific Batch Changes user journey in Settings but actually affects all subpages in Settings. Due to the number of components affected, I've left an estimate off of this one.

### Expected behavior

Something in the new settings content should be focused and the screen reader should dictate that that change occurred.

We're often waiting for the response to some GraphQL response before we have the exact content to show for those settings, but we should at least be able to focus the new settings area's header.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.

**Comment by umpox** [comment]
Fixed through https://github.com/sourcegraph/sourcegraph/pull/35413

All setting pages should now have landmark headings AFAICT, which allows quick navigation to this section
